dulwich/config.py

```diff
@@ -335,18 +356,26 @@ def get(  # type: ignore[override]
 
         return self._values[(section[0],)][name]
 
-    def set(self, section, name, value):
+    def set(
+        self,
+        section: KeyLike,
+        name: BytesLike,
+        value: BytesLike,
+    ) -> None:
         section, name = self._check_section_and_name(section, name)
 
-        if type(value) not in (bool, bytes):
+        if not isinstance(value, bytes):
```

Comment:
I decided that this was an accident and that the dictionary is not intended to hold booleans, say if that's wrong.

(There'd be some modest tweaks on the typing in this MR, and a couple of callers that should assert that they haven't got a surprise bool)